### PR TITLE
Added ability to edit and remove items from my closet

### PR DESCRIPTION
**Pull request by skoshkarli**
Added endpoints on the server side
Added db methods to handle the db functionality for the above

### CODE REVIEW

File: src/client/components/MyCloset/ItemModal.jsx [hunk 1]
```diff
@@ -1,19 +1,32 @@
...
 } from '../../actions/modalActions';
+import {
+  updateItemCategories,
+  updateItemColors,
+  updateItemBrands,
+} from '../../actions/myFilterActions';
 import { withRouter } from 'react-router-dom';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
+import {itemSeasonsExample} from "./ExampleData";
```
Comment (hcnimi) [inline src/client/components/MyCloset/ItemModal.jsx]: should we remove now that we're pulling from db?

File: src/client/components/MyCloset/ItemModal.jsx [hunk 2]
```diff
@@ -24,9 +37,80 @@ class ItemModal extends React.Component {
...
+      for(let i = 0; i < response.data.items[key].length; i++) {
+        itemsArray.push(response.data.items[key][i]);
+      }
+    }
+    return {response, itemsArray};
+  }
+
+  handleResponse(data) {
+    this.props.actions.updateItemCategories(data.response.data.categories);
+    //this.props.actions.updateItemSeasons(itemSeasonsExample); // temp
```
Comment (hcnimi) [inline src/client/components/MyCloset/ItemModal.jsx]: remove?

File: src/client/components/MyCloset/ItemModal.jsx [hunk 3]
```diff
@@ -64,14 +148,34 @@ class ItemModal extends React.Component {
...
   }
 
-  deleteItem() {
+  toggleDeleteButtonState() {
+    return this.props.actions.toggleDeleteButtonState(!this.props.deleteButtonState);
     //make server call to edit the item
   }
 
+  saveItem() {
+    //this.props.toggle({});
```
Comment (hcnimi) [inline src/client/components/MyCloset/ItemModal.jsx]: these will be needed later?

File: src/client/components/MyCloset/ItemModal.jsx (hunk 1 above)
Comment (skoshkarli) [reply src/client/components/MyCloset/ItemModal.jsx]: not seasons yet. thats one more thing which i have to tackle

File: src/client/components/MyCloset/ItemModal.jsx (hunk 2 above)
Comment (skoshkarli) [reply src/client/components/MyCloset/ItemModal.jsx]: leaving in for now for when i do the seasons

**Pull request merged**